AlicerceTests/Sources/Persistence/DiskMemoryPersistenceTestCase.swift

```diff
@@ -0,0 +1,235 @@
+//
+//  DiskMemoryPersistenceTestCase.swift
+//  Alicerce
+//
+//  Created by Luís Portela on 13/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import XCTest
+
+@testable import Alicerce
+
+final class DiskMemoryPersistenceTestCase: XCTestCase {
+
+    func testInit_UsingExtraPath_ItShouldCreateADirectory() {
+        let _ = diskMemoryPersistence(withDiskLimit: 0, memLimit: 0, extraPath: "TestInit")
+
+        let isDir = dirExists("TestInit")
+
+        XCTAssertTrue(isDir)
+
+        deleteItem("TestInit")
+    }
+
+    func testInit_WhenInitialisedWithLimit_ItShouldNotHoldMoreThanThoseFiles() {
+        let limit = mrMinderSize * 2
+        let diskMemPersistence = diskMemoryPersistence(withDiskLimit: limit, memLimit: 1, extraPath: "Test2")
+
+        let expectation1 = self.expectation(description: "Save MrMinder1")
+        persistMinder(with: "mr-minder", into: diskMemPersistence, expectation: expectation1)
+        let expectation2 = self.expectation(description: "Save MrMinder2")
+        persistMinder(with: "mr-minder1", into: diskMemPersistence, expectation: expectation2)
+        let expectation3 = self.expectation(description: "Save MrMinder3")
+        persistMinder(with: "mr-minder2", into: diskMemPersistence, expectation: expectation3)
+
+        func checkAndRetry(times: UInt, execute: @escaping (@escaping () -> Void) -> Void) {
+
+            func retry() {
+                Thread.sleep(forTimeInterval: 5)
+                if times > 0 {
+                    checkAndRetry(times: times - 1, execute: execute)
+                }
+            }
+
+            execute(retry)
+        }
+
+        waitForExpectations(timeout: 60) {
```

Comment:
Perhaps we could avoid these hardcoded expectations if we made the queues visible as `internal` on the `DiskMemoryPersistence`. 

That way we would simply schedule a block with the `expectation.fulfill()` on the right queue after scheduling the operations.

It would also not be visible to framework users since it would be `internal` 💪